services/backend/src/core/user/user.js

```diff
@@ -54,55 +58,82 @@ async function getUserById(request, response) {
         nameInitials: true,
         createdAt: true,
         updatedAt: true,
+        signupStep: true,
       },
-    })
-    .then((result) => response.status(200).json(result))
-    .catch((error) => {
-      console.log(error);
-      response.status(500).json("Unable to get profile");
     });
+
+    response.status(200).json(users);
+  } catch (error) {
+    console.log(error);
+    if (error.errors) {
+      response.status(422).json(error.errors);
+      return;
+    }
+    response.status(500).json("Unable to get profile");
+  }
 }
 
 async function createUser(request, response) {
   try {
-    const { body } = request;
+    validationResult(request).throw();
+
+    const { name, firstName, lastName, email } = request.body;
+
     const { id } = request.params;
     if (request.kauth.grant.access_token.content.sub === id) {
       const user = await prisma.users.create({
         data: {
           id,
-          name: body.name,
-          email: body.email,
+          name,
+          email,
+          firstName,
+          lastName,
           avatarColor: generateAvatarColor(),
-          nameInitials: `${body.firstName.charAt(0)}${body.lastName.charAt(0)}`,
+          nameInitials: `${firstName.charAt(0)}${lastName.charAt(0)}`,
           visibleCards: { create: {} },
         },
       });
+
       response.status(200).json(user);
+      return;
     }
     response
       .status(403)
       .json({ data: "Access to private account has be denied." });
   } catch (error) {
     console.log(error);
+    if (error.errors) {
+      response.status(422).json(error.errors);
+      return;
+    }
     response.status(500).json("Unable to create profiles");
   }
 }
 
 async function checkExistence(request, response) {
```

Comment:
I think this method will become useless since we have a sign up step created as soon as the user signs up. What do you think?

Reply:
yeah, I think so too, I'm not querying that endpoint while signing up, I just do a get, and catch it if it throws a 500, the do a post